### ISSUE TITLE
Quickfix to "MissingIncludes" adds absolute path to '#include' directive

### ISSUE THREAD

**Issue by SinixND** [clangd/clangd#2120]
**[Linking the associated stackoverflow post [here](https://stackoverflow.com/questions/78875589/add-missing-includes-without-path)]**

**Problem / Bug**
I am using clangd with VSCode. Clangd offers the possibility to add missing includes. When doing so, it (sometimes? couldn´t figure out when) automatically adds the relative or absolute path to the include directive. 

The latter causes problems when compiling on different OS's (in my case Linux and Termux (on my mobile)) as the absolute (system) paths are obviously different.

For example, instead of 

`#include "Header.h"`

it will add

`#include "/home/user/path/to/the/Header.h"`

**Question on stackoverflow**

Of course I can manually delete the path, but I wonder: 

Does someone know if there is a possibility to disable this behavior and make it omit the path?
I tried to google it, but unfortunately I couldn't find anything...

Thanks in advance

**Minimal example**

**Structure:**
```
./
- .clangd
- compile_commands.json
- bin/
- build/
- include/
--- Header.h
--- Subheader.h
- src/
--- main.cpp`
```
Contents:
```
// .clangd
Diagnostics:
  UnusedIncludes: Strict
  MissingIncludes: Strict
```

```
// Subheader.h
#ifndef IG20240815185417
#define IG20240815185417

struct Subheader {
    int i{2};
};

#endif
```
```
// Header.h
#ifndef IG20240520170135
#define IG20240520170135
    
#include "Subheader.h"
    
struct Header {
  Subheader sh{};
};

#endif
```
```
// main.cpp
#include "Header.h"

int main() {
  Header h{};
  Subheader sh{};
  return 0;
}
```

Compile command (with `compile_commands.json` creation):
```
clang++ -o build/main.o -c src/main.cpp -MJ build/main.o.json -I./include && clang++ -o bin/main.exe build/main.o && sed -e '1s/^/[\n/' -e '$s/,$/\n]/' build/*.o.json > compile_commands.json
```
Note that `compile_commands.json` will be created via the compile command. If it does not work on your machine, here is the basic compile command and the final content of the `.json` file. You will have to adjust the directory to match your system!

Compile command (without `compile_commands.json` creation):
```
clang++ -o build/main.o -c src/main.cpp -I./include && clang++ -o bin/main.exe build/main.o
```
```
// compile_commands.json
[
{ "directory": "/ABSOLUTE/PATH/TO/WORKING/DIRECTORY", "file": "src/main.cpp", "output": "build/main.o", "arguments": ["/usr/bin/clang++", "-xc++", "src/main.cpp", "-o", "build/main.o", "--driver-mode=g++", "-c", "-I", "./include", "--target=x86_64-pc-linux-gnu"]}
]
```

**Observation:**

When viewing `main.cpp` (in VSCode) it shows a blue squiggle line below `Subheader`, hinting that the respective header is missing / not directly included (but known via `Header.h`).

Note: When hovering the `Subheader` type keyword, the hover info panel states: `provided by: "/path/to/working/directory/include/Subheader.h"`.
Doing the same on `Header` states: `provided by: "Header.h"` 

When choosing the QuickFix option, it adds the include-directive like this:
`#include "/path/to/working/directory/include/Subheader.h"`

**Logs**
```
I[22:36:06.146] <-- textDocument/codeAction(39)
I[22:36:06.146] --> reply:textDocument/codeAction(39) 0 ms
I[22:36:06.146] --> textDocument/clangd.fileStatus
I[22:36:07.822] <-- textDocument/documentHighlight(40)
I[22:36:07.822] --> reply:textDocument/documentHighlight(40) 0 ms
I[22:36:07.822] --> textDocument/clangd.fileStatus
I[22:36:07.831] <-- textDocument/didChange
I[22:36:07.831] --> textDocument/clangd.fileStatus
I[22:36:07.855] <-- textDocument/codeAction(41)
I[22:36:07.855] ASTWorker building file /home/arch/coding/local_projects/cpp/test (copy 1)/src/main.cpp version 24 with command 
[/home/arch/coding/local_projects/cpp/test (copy 1)]
/usr/bin/clang++ -xc++ -o build/main.o --driver-mode=g++ -c -I ./include --target=x86_64-pc-linux-gnu -resource-dir=/home/arch/.config/Code/User/globalStorage/llvm-vs-code-extensions.vscode-clangd/install/18.1.3/clangd_18.1.3/lib/clang/18 -- "/home/arch/coding/local_projects/cpp/test (copy 1)/src/main.cpp"
I[22:36:07.857] --> textDocument/clangd.fileStatus
I[22:36:07.860] --> textDocument/clangd.fileStatus
I[22:36:07.860] --> textDocument/publishDiagnostics
I[22:36:07.860] --> textDocument/inactiveRegions
I[22:36:07.860] --> textDocument/clangd.fileStatus
I[22:36:07.861] --> reply:textDocument/codeAction(41) 5 ms
I[22:36:07.861] --> textDocument/clangd.fileStatus
I[22:36:07.862] Built preamble of size 224364 for file /home/arch/coding/local_projects/cpp/test (copy 1)/src/main.cpp version 24 in 0.00 seconds
I[22:36:07.862] --> workspace/semanticTokens/refresh(5)
I[22:36:07.862] --> textDocument/clangd.fileStatus
I[22:36:07.862] <-- reply(5)
I[22:36:07.865] <-- textDocument/documentHighlight(42)
I[22:36:07.867] --> textDocument/publishDiagnostics
I[22:36:07.867] --> textDocument/inactiveRegions
I[22:36:07.867] --> reply:textDocument/documentHighlight(42) 2 ms
I[22:36:07.867] --> textDocument/clangd.fileStatus
I[22:36:07.878] <-- textDocument/semanticTokens/full/delta(43)
I[22:36:07.878] --> reply:textDocument/semanticTokens/full/delta(43) 0 ms
I[22:36:07.878] --> textDocument/clangd.fileStatus
I[22:36:08.032] <-- textDocument/foldingRange(44)
I[22:36:08.032] --> reply:textDocument/foldingRange(44) 0 ms
I[22:36:08.129] <-- textDocument/codeAction(45)
I[22:36:08.129] --> reply:textDocument/codeAction(45) 0 ms
I[22:36:08.129] --> textDocument/clangd.fileStatus
I[22:36:08.184] <-- textDocument/documentSymbol(46)
I[22:36:08.184] --> reply:textDocument/documentSymbol(46) 0 ms
I[22:36:08.184] --> textDocument/clangd.fileStatus
I[22:36:08.196] <-- textDocument/documentSymbol(47)
I[22:36:08.196] --> reply:textDocument/documentSymbol(47) 0 ms
I[22:36:08.196] --> textDocument/clangd.fileStatus
I[22:36:08.833] <-- textDocument/documentLink(48)
I[22:36:08.833] --> reply:textDocument/documentLink(48) 0 ms
I[22:36:08.833] --> textDocument/clangd.fileStatus
I[22:36:09.105] <-- textDocument/inlayHint(49)
I[22:36:09.105] --> reply:textDocument/inlayHint(49) 0 ms
I[22:36:09.105] --> textDocument/clangd.fileStatus
```

**System information**

Output of `clangd --version`:
clangd version 18.1.8
Features: linux
Platform: x86_64-pc-linux-gnu

Editor/LSP plugin:
AUR: visual-studio-code-bin

Operating system:
Archlinux

**Comment by SinixND** [comment]
As per the suggestions provided via my (stackoverflow)[https://stackoverflow.com/questions/78875589/add-missing-includes-without-path/78894352#78894352] post:

I removed `UnusedIncludes: Strict` from `.clangd`, after which it won't show any hint or quickfix option at all. Therefore unfortunately this does note solve the issue.

Note: When hovering the `Subheader` type keyword, the hover info panel states: `provided by: "/path/to/working/directory/include/Subheader.h"`. Hovering over Header instead states: `provided by: "Header.h"`.

Edit: I just found out that removing the `MissingIncludes: Strict` line, but keeping the `UnusedIncludes: Strict` produces the expected results
